src/wallet/rpcdisclosure.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2017 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "base58.h"
+#include "rpcserver.h"
+#include "init.h"
+#include "main.h"
+#include "script/script.h"
+#include "script/standard.h"
+#include "sync.h"
+#include "util.h"
+#include "utiltime.h"
+#include "wallet.h"
+
+#include <fstream>
+#include <stdint.h>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/date_time/posix_time/posix_time.hpp>
+
+#include <univalue.h>
+
+using namespace std;
+
+void EnsureWalletIsUnlocked();
+bool EnsureWalletIsAvailable(bool avoidException);
+
+UniValue z_getpaymentdisclosure(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() != 1)
+        throw runtime_error(
+            "z_getpaymentdisclosure \"txid\" \"js_index\" \"output_index\" (\"message\") \n"
+            "\nGenerate a payment disclosure for a given joinsplit output.\n"
+            "\nEXPERIMENTAL FEATURE (TODO: ADD FLAG)\n"            
+            "\nArguments:\n"
+            "1. \"txid\"    (string, required) \n"
+            "2. \"js_index\"    (string, required) \n"
+            "3. \"output_index\"    (string, required) \n"
+            "4. \"message\"    (string, optional) \n"
+            "\nResult:\n"
+            "\"path\"           (string) The full path of the destination file\n"
```

Comment:
This PR is not ready for review, it's just a placeholder so we don't forget to complete proof of concept. The file path in the help string is left-over text copied from another rpc call...